app/ts/components/YearPicker.tsx

```diff
@@ -1,34 +1,39 @@
 import { Signal } from '@preact/signals'
-import * as funtypes from 'funtypes'
 
-interface CreateProps {
-	year: Signal<number>
-}
-
-export const YearPicker = ( { year }: CreateProps) => {
-	const setYear = (text: string) => {
-		const yearToSet = funtypes.Number.parse(text)
-		year.value = yearToSet
-	}
+export const YearPicker = ( { year, validYear }: { year: Signal<number>, validYear: Signal<boolean>  }) => {
 	const increment = () => {
 		year.value = year.value + 1
 	}
+
 	const decrement = () => {
 		year.value = Math.max(year.value - 1, 1)
 	}
-	return <div class = 'year-selector'>
-		<div class = 'year-selector-button year-selector-button-decrement' onClick = { decrement }>&ndash;</div>
-		<div class = 'year-selector-counter'>
-			<input
-				class = 'year-selector-counter-input'
-				maxlength = { 4 }
-				type = 'text'
-				pattern = '\d*'
-				value = { year.value }
-				onInput = { e => setYear(e.currentTarget.value) }
-			/>
-			<div class = 'year-selector-counter-num'>{ year }</div>
+
+	const handleInputChange = (e: Event) => {
+		const inputValue = (e.target as HTMLInputElement).value
```

Comment:
Three things here:
1. I recommend correctly typing the event so you don't have to cast.
2. I recommend using a more verbose variable name instead of `e`.
3. I am pretty sure you want `currentTarget` and not `target`.  You may want to investigate to verify though.
```suggestion
	const handleInputChange = (event: JSX.TargetedEvent<HTMLInputElement, Event>) => {
		const inputValue =event.currentTarget.value
```